src/design-system/components/Input/index.tsx

```diff
@@ -56,49 +45,124 @@ const ForwardInputRefFunction: React.ForwardRefRenderFunction<RNTextInput, TextI
     setIsFocused(false);
   };
 
-  const _onChangeText: ComponentProps<typeof RNTextInput>['onChangeText'] = value => {
+  const _onChangeText: ComponentProps<typeof RNTextInput>['onChangeText'] = newValue => {
     if (!onChangeText) return;
 
     if (pattern) {
-      const isValid = pattern.test(value);
-      onChangeText(isValid ? value : value.slice(0, -1));
+      const isValid = pattern.test(newValue);
+      onChangeText(isValid ? newValue : newValue.slice(0, -1));
     } else {
-      onChangeText(value);
+      onChangeText(newValue);
     }
   };
 
   const isInvalid = Boolean(error);
+  const showError = isInvalid && !hideError;
+  const isFilled = !!value;
 
   const defaultOnPressIcon = () => setToggle(prev => !prev);
 
+  const getCurrentColor = useCallback(
+    ({
+      disabled = 'n-600',
+      error: errorColor = 'error-500',
+      success = 'success-500',
+      focus = 'primary-500',
+      filled = 'n-600',
+      default: defaultColor = 'n-600',
+    }: DefaultColors = {}) => {
+      const disabledColor = isDisabled ? disabled : undefined;
+      const invalidColor = showError ? errorColor : undefined;
+      const successColor = isSuccess ? success : undefined;
+      const focusColor = isFocused ? focus : undefined;
+      const filledColor = isFilled ? filled : undefined;
+      const currentColor =
+        disabledColor ?? focusColor ?? invalidColor ?? successColor ?? filledColor ?? defaultColor;
+
+      return currentColor;
```

Comment:
Maybe is this what it doesn't like to code climate. Maybe we can try something like this?

```suggestion
if (isDisabled) return disabled;
if (showError) return errorColor;
if (isSuccess) return success;
if (isFocused) return focus;
if (isFilled) return filled;
return defaultColor;
```